core/src/test/java/kafka/admin/AclCommandTest.java

```diff
@@ -0,0 +1,589 @@
+    private static final Set<KafkaPrincipal> USERS = new HashSet<>(Arrays.asList(
+            SecurityUtils.parseKafkaPrincipal("User:CN=writeuser,OU=Unknown,O=Unknown,L=Unknown,ST=Unknown,C=Unknown"),
+            PRINCIPAL,
+            SecurityUtils.parseKafkaPrincipal("User:CN=\\#User with special chars in CN : (\\, \\+ \" \\ \\< \\> \\; ')")
+    ));
+    private static final Set<String> HOSTS = new HashSet<>(Arrays.asList("host1", "host2"));
+    private static final List<String> ALLOW_HOST_COMMAND = Arrays.asList("--allow-host", "host1", "--allow-host", "host2");
+    private static final List<String> DENY_HOST_COMMAND = Arrays.asList("--deny-host", "host1", "--deny-host", "host2");
+
+    private static final ResourcePattern CLUSTER_RESOURCE = new ResourcePattern(CLUSTER, Resource.CLUSTER_NAME, LITERAL);
+    private static final Set<ResourcePattern> TOPIC_RESOURCES = new HashSet<>(Arrays.asList(
+            new ResourcePattern(TOPIC, "test-1", LITERAL),
+            new ResourcePattern(TOPIC, "test-2", LITERAL)
+    ));
+    private static final Set<ResourcePattern> GROUP_RESOURCES = new HashSet<>(Arrays.asList(
+            new ResourcePattern(GROUP, "testGroup-1", LITERAL),
+            new ResourcePattern(GROUP, "testGroup-2", LITERAL)
+    ));
+    private static final Set<ResourcePattern> TRANSACTIONAL_ID_RESOURCES = new HashSet<>(Arrays.asList(
+            new ResourcePattern(TRANSACTIONAL_ID, "t0", LITERAL),
+            new ResourcePattern(TRANSACTIONAL_ID, "t1", LITERAL)
+    ));
+    private static final Set<ResourcePattern> TOKEN_RESOURCES = new HashSet<>(Arrays.asList(
+            new ResourcePattern(DELEGATION_TOKEN, "token1", LITERAL),
+            new ResourcePattern(DELEGATION_TOKEN, "token2", LITERAL)
+    ));
+    private static final Set<ResourcePattern> USER_RESOURCES = new HashSet<>(Arrays.asList(
+            new ResourcePattern(USER, "User:test-user1", LITERAL),
+            new ResourcePattern(USER, "User:test-user2", LITERAL)
+    ));
+
+    private static final Map<Set<ResourcePattern>, List<String>> RESOURCE_TO_COMMAND = new HashMap<Set<ResourcePattern>, List<String>>() {{
+            put(TOPIC_RESOURCES, Arrays.asList("--topic", "test-1", "--topic", "test-2"));
+            put(Collections.singleton(CLUSTER_RESOURCE), Collections.singletonList("--cluster"));
+            put(GROUP_RESOURCES, Arrays.asList("--group", "testGroup-1", "--group", "testGroup-2"));
+            put(TRANSACTIONAL_ID_RESOURCES, Arrays.asList("--transactional-id", "t0", "--transactional-id", "t1"));
+            put(TOKEN_RESOURCES, Arrays.asList("--delegation-token", "token1", "--delegation-token", "token2"));
+            put(USER_RESOURCES, Arrays.asList("--user-principal", "User:test-user1", "--user-principal", "User:test-user2"));
+        }};
+
+    private static final Map<Set<ResourcePattern>, Tuple2<Set<AclOperation>, List<String>>> RESOURCE_TO_OPERATIONS =
```

Comment:
Could you please use `Entry` to replace ``Tuple2``?

Reply:
Done